exp/api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -365,6 +368,74 @@ func (m *AzureManagedControlPlane) validateAPIServerAccessProfile() error {
 	return nil
 }
 
+// validateManagedClusterNetwork validates the Cluster network values.
+func (m *AzureManagedControlPlane) validateManagedClusterNetwork(cli client.Client) error {
+	ctx := context.Background()
+
+	// Fetch the Cluster.
+	clusterName, ok := m.Labels[clusterv1.ClusterLabelName]
+	if !ok {
+		return nil
+	}
+
+	ownerCluster := &clusterv1.Cluster{}
+	key := client.ObjectKey{
+		Namespace: m.Namespace,
+		Name:      clusterName,
+	}
+
+	if err := cli.Get(ctx, key, ownerCluster); err != nil {
+		if azure.ResourceNotFound(err) {
+			return nil
```

Comment:
That's a good point. I had copied the pattern from https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/55a16d82fb5759e44ca1e1fa8601f778a5f23b46/exp/api/v1beta1/azuremanagedmachinepool_webhook.go#L235 so we should make both consistent if we do decide to change this. I wonder if we should retry a couple of times before failing in case the AzureManagedCluster and the Cluster are created simultaneously. Do you know if kubectl apply -f template.yaml is serial? Will it block creation if the AzureManagedControlPlane is applied before the Cluster or will it proceed to create the Cluster? 🤔

Reply:
> so we should make both consistent if we do decide to change this.
+1 to making it consistent. I'm okay with merging this one as is, and create a followup issue to address this concern.

> I wonder if we should retry a couple of times before failing
I'm a bit hesitant about this because we don't want to add too much complexity in validation webhook.

> Will it block creation if the AzureManagedControlPlane is applied before the Cluster or will it proceed to create the Cluster?
I think it's serial. However, it wouldn't block creation if AzureManagedControlPlane is defined before Cluster in the yaml.

Reply:
I ran some tests and it definitely works fine and doesn't block creation even if AzureManagedControlPlane is defined before Cluster

Changed both to return an error if Cluster is not found